### PR TITLE
refactor(loadbalancer): migrate backend resource to plugin framework

### PR DESCRIPTION
**Pull request by villevsv-upcloud**

### CODE REVIEW

File: internal/service/loadbalancer/backend.go [hunk 1]
```diff
@@ -4,366 +4,506 @@ import (
...
 			},
 		},
+		Version: 1,
 	}
 }
 
-func resourceBackendCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) (diags diag.Diagnostics) {
-	svc := meta.(*service.Service)
-	serviceID := d.Get("loadbalancer").(string)
+func (r *backendResource) ModifyPlan(ctx context.Context, req resource.ModifyPlanRequest, resp *resource.ModifyPlanResponse) {
```
Comment (kangasta) [inline internal/service/loadbalancer/backend.go:264]: Is this needed? Looks like `p` is not used after getting an attribute value to it.

File: internal/service/loadbalancer/backend.go [hunk 2]
```diff
@@ -4,366 +4,506 @@ import (
...
+							Computed:            true,
+							Default:             booldefault.StaticBool(false),
+						},
+						"tls_verify": schema.BoolAttribute{
+							MarkdownDescription: "Enables backend servers certificate verification. Please make sure that TLS config with the certificate bundle of type authority attached to the backend or `tls_use_system_ca` enabled. Note: `tls_verify` has preference over `health_check_tls_verify` when `tls_enabled` in true.",
+							Optional:            true,
+							Computed:            true,
+							Default:             booldefault.StaticBool(false),
+						},
+					},
```
Comment (kangasta) [inline internal/service/loadbalancer/backend.go:253]: There used to be read-only `tls_configs` property (probably duplicate of main level `tls_configs` field), could add CHANGELOG note about its removal 🤔

**Review by kangasta — APPROVED** [review]

File: internal/service/loadbalancer/backend.go (hunk 1 above)
Comment (villevsv-upcloud) [reply internal/service/loadbalancer/backend.go]: Removed as unnecessary.

File: internal/service/loadbalancer/backend.go (hunk 2 above)
Comment (villevsv-upcloud) [reply internal/service/loadbalancer/backend.go:252]: Added a note.

**Pull request merged**